rclpy/src/rclpy/detail/logging_mutex.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "./logging_mutex.hpp"
+
+#include <memory>
+#include <mutex>
+
+#include "rcutils/macros.h"
+
+std::shared_ptr<std::recursive_mutex>
```

Comment:
nit: we could make use of a `std::recursive_mutex` directly, I don't see an advantage of using `std::shared_ptr` here (all Python calls will happen after "library construction" and before "library destruction").

Reply:
Yeah I thought about this and considered changing it, but I also don't think it is really a problem to use a shared pointer here.

I can change it though.